service/worker/indexer/esProcessor.go

```diff
@@ -266,6 +266,25 @@ func (p *ESProcessorImpl) shadowBulkAfterAction(id int64, requests []bulk.Generi
 				tag.ESResponseStatus(err.Status),
 				tag.ESRequest(request.String()))
 		}
+		return
+	}
+	responseItems := response.Items
+	for i := 0; i < len(requests); i++ {
+		key := p.retrieveKafkaKey(requests[i])
+		if key == "" {
+			continue
+		}
+		responseItem := responseItems[i]
```

Comment:
just to be on the safe side let's check if `i` is greater than `len(responseItems)`